sources/packages/backend/apps/api/src/route-controllers/student-appeal/_tests_/e2e/student-appeal.students.controller.submitStudentAppeal.e2e-spec.ts

```diff
@@ -0,0 +1,560 @@
+import { HttpStatus, INestApplication } from "@nestjs/common";
+import * as request from "supertest";
+import { DataSource, Repository } from "typeorm";
+import { TestingModule } from "@nestjs/testing";
+import {
+  BEARER_AUTH_TYPE,
+  createTestingAppModule,
+  FakeStudentUsersTypes,
+  getStudentToken,
+  mockUserLoginInfo,
+} from "../../../../testHelpers";
+import {
+  createFakeStudentAppeal,
+  createFakeStudentAppealRequest,
+  getProviderInstanceForModule,
+  saveFakeApplication,
+  saveFakeStudent,
+} from "@sims/test-utils";
+import {
+  Application,
+  ApplicationStatus,
+  Student,
+  StudentAppeal,
+  StudentAppealRequest,
+  StudentAppealStatus,
+} from "@sims/sims-db";
+import {
+  APPLICATION_CHANGE_NOT_ELIGIBLE,
+  INVALID_APPLICATION_NUMBER,
+} from "../../../../constants";
+import { FormService } from "../../../../services";
+import { AppStudentsModule } from "../../../../app.students.module";
+
+describe("StudentAppealStudentsController(e2e)-submitStudentAppeal", () => {
+  let app: INestApplication;
+  let appDataSource: DataSource;
+  let appModule: TestingModule;
+  let student: Student;
+  let studentToken: string;
+  let applicationRepo: Repository<Application>;
+  let studentAppealRequestRepo: Repository<StudentAppealRequest>;
+  let studentAppealRepo: Repository<StudentAppeal>;
+  let formService: FormService;
+
+  beforeAll(async () => {
+    const { nestApplication, module, dataSource } =
+      await createTestingAppModule();
+    app = nestApplication;
+    appDataSource = dataSource;
+    appModule = module;
+    applicationRepo = dataSource.getRepository(Application);
+    studentAppealRequestRepo = dataSource.getRepository(StudentAppealRequest);
+    studentAppealRepo = dataSource.getRepository(StudentAppeal);
+    student = await saveFakeStudent(appDataSource);
```

Comment:
I would not recommend using same student for all tests as we extend the scenarios, there could be data interference.

Reply:
In my current PR, I also used a student globally defined but I made that explicit naming it as `sharedStudent`. Any test that would demand customization would need to create a new one.

Reply:
I also see the token being generated at before each here. We had suggestions in the past to keep it in test level right?